django_db_schema/hospital/models.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import models
+
+
+# Create your models here.
+class Hospital(models.Model):
+    name = models.CharField(max_length=100)
+    city = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.name
+
+
+class Department(models.Model):
+    hospital = models.ForeignKey(Hospital, on_delete=models.CASCADE)
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.name
+
+
+class Person(models.Model):
+    name = models.CharField(max_length=100)
+    age = models.PositiveSmallIntegerField()
+    gender = models.CharField(max_length=1, choices=[('M', 'Male'), ('F', 'Female'), ('O', 'Other')])
+
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return self.name
+
+
+class Worker(Person):
```

Comment:
List the pros and cons in inheriting `Person` VS having foreignkey to `Person`

Reply:
pros:
A worker could be also a patient, therefore if it will be managed in another table, the data will not be saved twice. 
Each change in Person class will change all the tables inheriting from it.
cons:
To save a worker we will need to save 2 objects: the Person and the Worker.
It will be more difficult to change a Worker instance if the data is splitted between tables

Reply:
nice
also, person can have multiple instances of workers which will make it even more difficult to manage
I suggest to change it from inheritance to foreignkey
by the way, if person couldn't have multiple workers, `OneToOne` relation would be the solution

Reply:
d